### ISSUE TITLE
Incorrect Polygons for some filled contour plots

### ISSUE THREAD

**Issue by mikewalker92** [SciTools/cartopy#310]
The following code does not produce the desired plot.
Instead, this returns a green envelope shape spanning the entire map. 

```
import iris
import matplotlib.pyplot as plt
import iris.quickplot as qplt

fname = iris.sample_data_path('uk_hires.pp')
cubes = iris.load(fname)
cube = cubes[0]

sub_cube = cube[2,5]

qplt.contourf(sub_cube, 43)
plt.show()
```

Plotting with less than 43 contours returns the expected results.
Plotting with different choices of sub cubes also produces the expected results.

**Comment by pelson** [comment]
Looks like there are a number of problems with the geometries produced in this contour. One example:

```
from matplotlib.path import Path
import numpy as np
import matplotlib.pyplot as plt
import matplotlib.patches as mpatches
import cartopy
import cartopy.crs as ccrs

sphere = ccrs.Globe(semimajor_axis=6371229.0,
                    semiminor_axis=6371229.0,
                    ellipse=None)
rp = ccrs.RotatedPole(177.5, 37.5, globe=sphere)
rp_geod = ccrs.RotatedGeodetic(177.5, 37.5, globe=sphere)

bad_path = Path._fast_from_codes_and_verts(np.array([[ 359.58648681640625  ,    1.6967999935150084],
       [ 359.58648681640625  ,    1.6967999935150146],
       [ 359.5999755859375   ,    1.7103000879287658],
       [ 359.5999755859375   ,    1.710300087928772 ],
       [ 359.5999755859375   ,    1.7238000631332397],
       [ 359.5999755859375   ,    1.7373000383377075],
       [ 359.5999755859375   ,    1.7508000135421753],
       [ 359.613494873046875 ,    1.7642999887466368],
       [ 359.613494873046875 ,    1.7642999887466431],
       [ 359.613494873046875 ,    1.7778000831604004],
       [ 359.613494873046875 ,    1.7778000831604066],
       [ 359.5999755859375   ,    1.7778000831604035],
       [ 359.5999755859375   ,    1.7778000831604004],
       [ 359.5999755859375   ,    1.7642999887466431],
       [ 359.5999755859375   ,    1.7508000135421753],
       [ 359.58648681640625  ,    1.7373000383377137],
       [ 359.58648681640625  ,    1.7373000383377075],
       [ 359.58648681640625  ,    1.7373000383377013],
       [ 359.5999755859375   ,    1.7238000631332397],
       [ 359.58648681640625  ,    1.7103000879287782],
       [ 359.58648681640625  ,    1.710300087928772 ],
       [ 359.58648681640625  ,    1.6967999935150146],
       [ 359.58648681640625  ,    1.6967999935150084]]),
            np.array([1, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2], dtype=np.uint8))

ax = plt.axes(projection=rp)
patch = mpatches.PathPatch(bad_path, facecolor='red', alpha=0.2, transform=rp_geod)
ax.add_patch(patch)
ax.autoscale_view()
plt.show()
```

![figure_1](https://cloud.githubusercontent.com/assets/810663/8426910/3a895bc0-1f0b-11e5-8bb5-88b275bb5092.png)

**Comment by pelson** [comment]
Path reduced to:

```
verts = np.array([[ 359.5  ,    1.6967999935150084],
                  [ 359.5  ,    1.6967999935150146],
                  [ 359.613494873046875 ,    1.7642999887466368],
                  [ 359.613494873046875 ,    1.7642999887466431],
                  [ 359.5920764726345737,    1.7421173469387756],
                  [ 359.5  ,    1.6967999935150146],
                  [ 359.5  ,    1.6967999935150084]])
codes = [1, 2, 2, 2, 2, 2, 2]
bad_path = Path(verts, codes)
```

Seems highly likely that this is a threshold issue - changing some of the floats makes this case work again.

**Comment by pelson** [comment]
Whilst some of the broken down problems are resolved in ``v0.16``, the original issue remains.

**Comment by QuLogic** [comment]
This seems fixed on `master`.
